rotkehlchen/chain/ethereum/modules/aave/aave.py

```diff
@@ -163,6 +176,190 @@ def _get_balances(
 
         return aave_balances
 
+    def _calculate_loss(
+            self,
+            user_address: ChecksumEvmAddress,
+            events: list[EvmEvent],
+            balances: AaveBalances,
+    ) -> tuple[dict[Asset, Balance], dict[Asset, Balance], dict[Asset, Balance]]:
+        """
+        Returns a tuple of mapping of losses due to liquidation/borrowing and
+        earnings due to keeping the principal repaid by the liquidation
+        """
+        historical_borrow_balances: dict[Asset, FVal] = defaultdict(FVal)
+        total_lost: dict[Asset, Balance] = defaultdict(Balance)
+        total_earned: dict[Asset, Balance] = defaultdict(Balance)
+        atokens_balances: dict[Asset, Balance] = defaultdict(Balance)
+        earned_atoken_balances: dict[Asset, Balance] = defaultdict(Balance)
+        prev_balance = Balance()
+
+        for event in events:
+            if event.asset in earned_atoken_balances:
+                prev_balance = earned_atoken_balances[event.asset]
+
+            if event.event_subtype in (HistoryEventSubType.GENERATE_DEBT, HistoryEventSubType.DEPOSIT_ASSET):  # noqa: E501
+                historical_borrow_balances[event.asset] -= event.balance.amount
+            elif event.event_subtype in (HistoryEventSubType.PAYBACK_DEBT, HistoryEventSubType.REMOVE_ASSET, HistoryEventSubType.REWARD):  # noqa: E501
+                if event.extra_data is not None and 'is_liquidation' in event.extra_data:
+                    total_earned[event.asset] += event.balance
+                if event.event_subtype == HistoryEventSubType.REWARD:
```

Comment:
No but the if statament was meant to change where the asset is added in the else part. If is reward put it here otherwise add to the historical borrow balances. Will change a bit